### ISSUE TITLE
NotImplementedError: C extensions are not supported on upgrading puma from 2.14.0 to 2.15.2 on jruby

### ISSUE THREAD

**Issue by nikita-barsukov** [puma/puma#821]
I'm trying to upgrade puma in my rails application from 2.14.0 to 2.15.2 using bundler. JRuby version I use is 9.0.3.0

When I run `bundle update puma` I get an error about native C extensions. Is this something that can be fixed?

Stacktrace I get: 

```
Installing puma 2.15.2 (was 2.14.0) with native extensions

Gem::Ext::BuildError: ERROR: Failed to build gem native extension.

    /home/nikita/.rbenv/versions/jruby-9.0.3.0/bin/jruby -r ./siteconf20151107-8495-1mk0snj.rb extconf.rb
NotImplementedError: C extensions are not supported
    <top> at /home/nikita/.rbenv/versions/jruby-9.0.3.0/lib/ruby/stdlib/mkmf.rb:1
  require at org/jruby/RubyKernel.java:939
   (root) at /home/nikita/.rbenv/versions/jruby-9.0.3.0/lib/ruby/stdlib/rubygems/core_ext/kernel_require.rb:1
    <top> at extconf.rb:1

extconf failed, exit code 1
```

**Comment by nikita-barsukov** [comment]
#238 looks related

**Comment by evanphx** [comment]
Sorry about that! I've just released 2.15.3-java properly!
